src/include/processor/execution_context.h

```diff
@@ -1,17 +1,26 @@
 #pragma once
 
 #include "common/profiler.h"
+#include "common/types/uuid.h"
 #include "main/client_context.h"
 
 namespace kuzu {
 namespace processor {
 
 struct ExecutionContext {
+    std::string queryId;
     common::Profiler* profiler;
     main::ClientContext* clientContext;
 
-    ExecutionContext(common::Profiler* profiler, main::ClientContext* clientContext)
-        : profiler{profiler}, clientContext{clientContext} {}
+    ExecutionContext(common::Profiler* profiler, main::ClientContext* clientContext, std::string id)
+        : profiler{profiler}, clientContext{clientContext} {
```

Comment:
I think @ray6080 will need a system level query id soon. So let's make this id generated from database which you will need a counter and a lock. Then ExecutionContext should always be created with an id and does not need to generate one in the constructor.


Also I would suggest using a integer based id.